cmake/common.cmake

```diff
@@ -0,0 +1,165 @@
+    set(TARGET_SOC          J721E)
+elseif ("${TARGET_SOC_LOWER}" STREQUAL "j721s2")
+    set(TARGET_PLATFORM     J7)
+    set(TARGET_CPU          A72)
+    set(TARGET_OS           LINUX)
+    set(TARGET_SOC          J721S2)
+elseif ("${TARGET_SOC_LOWER}" STREQUAL "j784s4")
+    set(TARGET_PLATFORM     J7)
+    set(TARGET_CPU          A72)
+    set(TARGET_OS           LINUX)
+    set(TARGET_SOC          J784S4)
+elseif ("${TARGET_SOC_LOWER}" STREQUAL "am62a")
+    set(TARGET_PLATFORM     SITARA)
+    set(TARGET_CPU          A53)
+    set(TARGET_OS           LINUX)
+    set(TARGET_SOC          AM62A)
+else()
+    message(FATAL_ERROR "SOC ${TARGET_SOC_LOWER} is not supported.")
+endif()
+
+message("SOC=${TARGET_SOC_LOWER}")
+
+add_definitions(
+    -DTARGET_CPU=${TARGET_CPU}
+    -DTARGET_OS=${TARGET_OS}
+    -DSOC_${TARGET_SOC}
+)
+
+set(DLPACK_INSTALL_DIR ${TARGET_FS}/usr/include/dlpack)
+
+link_directories(${TARGET_FS}/usr/lib/aarch64-linux-gnu
+                 ${TARGET_FS}/usr/lib/
+                 )
+
+#message("PROJECT_SOURCE_DIR =" ${PROJECT_SOURCE_DIR})
+#message("CMAKE_SOURCE_DIR =" ${CMAKE_SOURCE_DIR})
+
+include_directories(${PROJECT_SOURCE_DIR}
+                    ${PROJECT_SOURCE_DIR}/..
+                    ${PROJECT_SOURCE_DIR}/include
+                    SYSTEM ${TARGET_FS}/usr/local/include
+                    SYSTEM ${TARGET_FS}/usr/include/processor_sdk/vision_apps
+                    SYSTEM ${TARGET_FS}/usr/include/edgeai-tiovx-modules
+                    SYSTEM ${TARGET_FS}/usr/include/processor_sdk/tiovx/include
+                    SYSTEM ${TARGET_FS}/usr/include/processor_sdk/tiovx/kernels/include/
+                    SYSTEM ${TARGET_FS}/usr/include/processor_sdk/tiovx/kernels_j7/include/
+                    SYSTEM ${TARGET_FS}/usr/include/processor_sdk/vision_apps/kernels/img_proc/include/
+                    SYSTEM ${TARGET_FS}/usr/include/processor_sdk/tidl_j7/ti_dl/inc
+                    SYSTEM ${TARGET_FS}/usr/include/processor_sdk/ivision
+                    SYSTEM ${TARGET_FS}/usr/include/processor_sdk/tiovx/utils/include
+                    SYSTEM ${TARGET_FS}/usr/include/edgeai-tiovx-kernels/
+                    SYSTEM ${TARGET_FS}/usr/include/processor_sdk/vision_apps/kernels/fileio/include/
+                    SYSTEM ${TARGET_FS}/usr/include/processor_sdk/imaging/kernels/include/
+                    SYSTEM ${TARGET_FS}/usr/include/processor_sdk/app_utils/
+                    SYSTEM ${TARGET_FS}/usr/include/processor_sdk/vision_apps/utils/app_init/include/
+                    SYSTEM ${TARGET_FS}/usr/include/edgeai-apps-utils
+                    SYSTEM ${TARGET_FS}/usr/include/processor_sdk/vision_apps/kernels/fileio/include/
+                    SYSTEM ${TARGET_FS}/usr/include/processor_sdk/video_io/kernels/include/
+                    SYSTEM ${TARGET_FS}/usr/include/processor_sdk/imaging
+                    SYSTEM ${TARGET_FS}/usr/include/processor_sdk/imaging/kernels/include/
```

Comment:
This folder is listed twice in this list

Reply:
Removed it now